src/ess.jl

```diff
@@ -238,7 +238,7 @@ function ess_rhat(
     rhat = Vector{T}(undef, nparams)
 
     # for each parameter
-    for (i, chains_slice) in enumerate((view(chains, :, i, :) for i in axes(chains, 2)))
+    for (i, chains_slice) in enumerate((selectdim(chains, 1, i) for i in axes(chains, 1)))
```

Comment:
Isn't that just
```suggestion
    for (i, chains_slice) in enumerate(eachslice(chains; dims=1))
```
?

Reply:
`eachslice` isn't present in Julia 1.0. We could add Compat as a direct dependency (it's already an indirect one) to use it.

Reply:
If it's an indirect one anyway, then maybe let's use `eachslice` + Compat? I think that's better in the long run since I assume we won't remember to update these lines when we drop support for older Julia versions.